lighthouse-core/gather/gatherers/scripts.js

```diff
@@ -0,0 +1,135 @@
+    // Go read the comments in network-recorder.js _findRealRequestAndSetSession.
+    let sessionId = event.sessionId;
+    if (this._mainSessionId === null) {
+      this._mainSessionId = sessionId;
+    }
+    if (this._mainSessionId === sessionId) {
+      sessionId = undefined;
+    }
+
+    // We want to ignore scripts from OOPIFs. In reality, this does more than block just OOPIFs,
+    // it also blocks scripts from the same origin but that happen to run in a different process,
+    // like a worker.
+    if (event.method === 'Debugger.scriptParsed' && !sessionId) {
+      // Events without an embedderName (read: a url) are for JS that we ran over the protocol.
+      if (event.params.embedderName) this._scriptParsedEvents.push(event.params);
+    }
+  }
+
+  /**
+   * @param {LH.Gatherer.FRTransitionalContext} context
+   */
+  async startInstrumentation(context) {
+    const session = context.driver.defaultSession;
+    session.addProtocolMessageListener(this.onProtocolMessage);
+    await session.sendCommand('Debugger.enable');
+  }
+
+  /**
+   * @param {LH.Gatherer.FRTransitionalContext} context
+   */
+  async stopInstrumentation(context) {
+    const session = context.driver.defaultSession;
+    const formFactor = context.baseArtifacts.HostFormFactor;
+
+    session.removeProtocolMessageListener(this.onProtocolMessage);
+
+    // Without this line the Debugger domain will be off in FR runner.
+    // Odd, because `startInstrumentation` enabled it...
+    await session.sendCommand('Debugger.enable');
+
+    // If run on a mobile device, be sensitive to memory limitations and only
+    // request one at a time.
+    this._scriptContents = await runInSeriesOrParallel(
+      this._scriptParsedEvents,
+      ({scriptId}) => {
+        return session.sendCommand('Debugger.getScriptSource', {scriptId})
+          .then((resp) => resp.scriptSource)
+          .catch(() => undefined);
+      },
+      formFactor === 'mobile' /* runInSeries */
+    );
+    await session.sendCommand('Debugger.disable');
+  }
+
+  async getArtifact() {
+    /** @type {LH.Artifacts['Scripts']} */
+    const scripts = this._scriptParsedEvents.map((event, i) => {
+      // 'embedderName' and 'url' are confusingly named, so we rewrite them here.
+      // On the protocol, 'embedderName' always refers to the URL of the script (or HTML if inline).
+      // Same for 'url' ... except, magic "sourceURL=" comments will override the value.
+      // It's nice to display the user-provided value in Lighthouse, so we add a field 'name'
+      // to make it clear this is for presentational purposes.
+      // See https://chromium-review.googlesource.com/c/v8/v8/+/2317310
+      return {
+        name: event.url,
+        ...event,
+        // embedderName is optional on the protocol because backends like Node may not set it.
+        // For our purposes, it is always set. But just in case it isn't... fallback to the url.
+        url: event.embedderName || event.url,
```

Comment:
Should this be `??`? If `embedderName` is empty wouldn't we want to keep it that way?

Reply:
nope, that's the "no idea if this can happen" edge case I want to avoid.